File: golang/1657.go
func closeStrings(word1 string, word2 string) bool {
	if len(word1) != len(word2) {
		return false
	}


	m := make(map[int32]int)
	n := make(map[int32]int)

	for _, v := range word1 {
		m[v]++
	}
	for _, v := range word2 {
		n[v]++
	}

	var mk []int
	var mv []int

	var nk []int
	var nv []int

	for k, v := range m {
		mk = append(mk, int(k))
		mv = append(mv, v)
	}
	for k, v := range n {
		nk = append(nk, int(k))
		nv = append(nv, v)
	}

	sort.Ints(mk)
	sort.Ints(mv)
	sort.Ints(nk)
	sort.Ints(nv)

	return slices.Equal(mk, nk) && slices.Equal(mv, nv)
}

File: rust/1657.rs
impl Solution {
    pub fn close_strings(word1: String, word2: String) -> bool {
        if word1.len() != word2.len() {
            return false;
        }
        use std::collections::HashMap;
        let mut m1 = HashMap::new();
        let mut m2 = HashMap::new();
        for c in word1.chars() {
            *m1.entry(c).or_insert(0) += 1;
        }
        for c in word2.chars() {
            *m2.entry(c).or_insert(0) += 1;
        }

        let mut v1: Vec<_> = m1.values().collect();
        let mut v2: Vec<_> = m2.values().collect();

        let mut k1: Vec<_> = m1.keys().collect();
        let mut k2: Vec<_> = m2.keys().collect();
        k1.sort();
        k2.sort();
        if k1 != k2 {
            return false;
        }

        v1.sort();
        v2.sort();

        v1 == v2 
    }
}
